### ISSUE TITLE
Add dojoConfig.modulePaths to page only when needed

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#1235]
@jhpedemonte

We should probably avoid writing out the 'widgets' modulePaths to the file unless the custom widget feature is used... in many cases, that would mean no modulePaths needed at all.  It complicates the source.  You might try just mixing it in to Context.prototype._configProps as you use the custom widgets feature...  and we would also need to take that out of the prototype

Also, the file segment in modulePaths is "undefined/lib/widgets" for me

**Comment by peller** [comment]
also, config.modulePaths should be replaced by config.packages

**Comment by peller** [comment]
see @jhpedemonte 's commit: e2100298f137098b367218b5a2b0d087821d3d45

**Comment by childsb** [comment]
@peller no longer using modulePath correct? close?

**Comment by peller** [comment]
no, we're using `packages` instead.  We shouldn't be adding references to the 'widgets' stuff unless it's being used.  An empty page shouldn't have unused references like that. Just going to confuse the user.

**Comment by jhpedemonte** [comment]
There's a place in the code (that I added) which runs code when the first widget from a library is added to page.  That would be the correct place to do this.

I don't think we need this for M4 right now.  There'll be some cruft in the **dojo-config** line, but I think that's OK for now.

I'll take this bug, but for M5.  Let me know if you disagree...

**Comment by peller** [comment]
I think we're better off without the cruft, for migration purposes, etc.  Better perhaps to comment out the line that adds the 'widgets' in M4 and put it back in M5?

**Comment by peller** [comment]
@JonFerraiolo

**Comment by jhpedemonte** [comment]
Well, it's just not **"widgets"**, it's also the other libraries.  They're all added to that line, whether you use their widgets or not.

**Comment by peller** [comment]
only libraries which are set up for that page.  "widgets" is going to propagate to all user-created files starting in M4, and I don't think that's good.

**Comment by peller** [comment]
yes, so after the script-loading-redo merge, we now have declarations for jquery, yui, etc. also, which weren't on pages before.  I guess this is more tolerable in the interim, since it can be controlled by the "Modify Libraries" setting.

**Comment by jhpedemonte** [comment]
-> M6

**Comment by jhpedemonte** [comment]
-> M7
